diamond-api/src/find.rs

```diff
@@ -218,10 +227,7 @@ fn walk_tree(
     Ok(metrics)
 }
 
-pub fn find_handler(
-    ctx: Data<Context>,
-    query: FindQuery,
-) -> impl Future<Item = HttpResponse, Error = Error> {
+pub async fn find_handler(ctx: Data<Context>, query: FindQuery) -> Result<HttpResponse, Error> {
     match FindPath::from(&query) {
```

Comment:
`let path = FindPath::from(&query).map_err(ErrorInternalServerError)?;`

Reply:
This is an old code, but good catch